api/src/test/java/org/openmrs/util/databasechange/Database1_9_7UpgradeIT.java

```diff
@@ -247,7 +246,7 @@ public void shouldFailIfAnyDrugOrderFrequenciesNotMappedToConceptsAreFound() thr
 		String errorMsgSubString1 = "liquibase.exception.MigrationFailedException: Migration failed for change set liquibase-update-to-latest.xml::201401101647-TRUNK-4187::wyclif";
 		
 		IOException exception = assertThrows(IOException.class, () -> upgradeTestUtil.upgrade());
-		assertThat(exception.getMessage(), is(errorMsgSubString1));
+		assertThat(exception.getMessage(), startsWith(errorMsgSubString1));
```

Comment:
I thought `is` and `startsWith` are different. Correct?

Reply:
For instance: `This is a` vs `This is a boy`

Reply:
Yes they are :) `is` checks for equality as opposed to `startsWith`
The thing is that the `expectedException.expectMessage()` was doing a `containsString` so sometimes the string we expect was an exact match, sometimes in was not. In the former case using the `is` is fine, in the later we have to either use `startsWith` or `containsString`

Reply:
Is line 250 above also using expectedException.expectMessage()?

Reply:
Not anymore. We are replacing the deprecated Junit 4 ExpectedException. Here you can see the first change https://github.com/openmrs/openmrs-core/commit/d0871eeddc746fe858be9680c273cf62d58e95e1#diff-9715472d14ed569e3b08d1dfaafec969L251-R250 which failed since we used the `is` assertion but `expectedException.expectMessage()` does a `containsString` which in this case made a difference since `errorMsgSubString1` is just a substring.

Reply:
Oh i see!